include/deal.II/lac/arpack_solver.h

```diff
@@ -139,13 +157,20 @@ class ArpackSolver : public Subscriptor
 
   /**
    * Solve the generalized eigensprectrum problem $A x=\lambda B x$ by calling
-   * the <code>dneupd</code> and <code>dnaupd</code> functions of ARPACK.
+   * the <code>dsaupd</code> and <code>dseupd</code> or
+   * <code>dnaupd</code> and <code>dneupd</code> functions of ARPACK.
    *
    * The function returns a vector of eigenvalues of length <i>n</i> and a
-   * vector of eigenvectors, where the latter should be twice the size of the
-   * eigenvalue vector. The first <i>n</i> vectors in
-   * <code>eigenvectors</code> will be the real parts of the eigenvectors, the
-   * second <i>n</i> the imaginary parts.
+   * vector of eigenvectors of length <i>n</i> in the symmetric case
+   * and of length <i>n+1<i> in the non-symmetric case. In the symmetric case
```

Comment:
I think the critical test case is exactly when we need n+1 real-valued vectors to store all the information and also arpack needs to store n+1 eigenvalues, which is exactly when n and n+1 form a complex conjugate pair. Thats exactly the case in the test example with eigenvalues

(32.2359,0.00000)
(40.3938,-4.48216)
(40.3938,4.48216)
(50.5706,-9.07560)

where one complex pair is completely included but the last one belongs also to a complex pair where we only asked for the first one.